fast64_internal/oot/c_writer/oot_scene_table_c.py

```diff
@@ -1,75 +1,211 @@
+	try:
+		with open(os.path.join(exportPath, 'include/tables/scene_table.h')) as fileData:
+			# keep the relevant data and do some formatting
+			for i, line in enumerate(fileData):
+				if not line.startswith("// "):
+					if not (line.startswith("/**") or line.startswith(" *")):
+						dataList.append(
+							line[(line.find("(") + 1) :]
+							.rstrip(")\n")
+							.replace(" ", "")
+							.split(",")
+						)
+					else:
+						fileHeader += line
+				if line.startswith("/* 0x"):
+					startIndex = line.find("SCENE_")
+					sceneNames.append(line[startIndex : line.find(",", startIndex)])
+	except FileNotFoundError:
+		raise PluginError("ERROR: Can't find scene_table.h!")
+
+	# return the parsed data, the header comment and the comment mentionning debug scenes
+	return dataList, fileHeader, sceneNames
+
+def getSceneIndex(sceneNameList, sceneName):
+	'''Returns the index (int) of the chosen scene, returns None if ``Custom`` is chosen'''
+	if sceneName == "Custom":
+		return None
+
+	if sceneNameList is not None:
+		for i in range(len(sceneNameList)):
+			if sceneNameList[i] == sceneName:
+				return i
+
+	# intended return value to check if the chosen scene was removed
 	return None
 
-def readSceneTable(exportPath):
-	fileData = readFile(os.path.join(exportPath, 'src/code/z_scene_table.c'))
-
-	matchResult = re.search('SceneTableEntry\s*gSceneTable\[\]\s*=\s*\{([^\}]*)\}', fileData, re.DOTALL)
-	if matchResult is None:
-		raise PluginError("z_scene_table.c does not have gSceneTable in it.")
-	sceneTable = parseSceneTableData(matchResult.group(1))
-
-	return sceneTable, fileData, matchResult.start(0), matchResult.end(0)
-
-def parseSceneTableData(data):
-	table = []
-	for match in re.finditer('([^,]*)\s*\(([^,]*),\s*([^,]*),\s*([^,]*),\s*([^,]*)(,\s*([^,]*))?\)\s*,', data):
-		name = match.group(2)
-		sceneType = match.group(1).strip()
-		if sceneType == "UNTITLED_SCENE":
-			table.append(SceneTableEntry(sceneType, name, None, match.group(3), match.group(4), match.group(5)))
-		elif sceneType == "TITLED_SCENE":
-			table.append(SceneTableEntry(sceneType, name, match.group(3), match.group(4), match.group(5), match.group(7)))
-		else:
-			raise PluginError("Unhandled scene entry type: " + str(sceneType))
-	return table
+def getOriginalIndex(sceneName):
+	'''
+		Returns the index of a specific scene defined by which one the user chose
+		or by the ``sceneName`` parameter if it's not set to ``None``
+	'''
+	i = 0
+
+	if sceneName != "Custom":
+		for elem in ootEnumSceneID:
+			if elem[0] == sceneName:
+				# returns i - 1 because the first entry is the ``Custom`` option
+				return i - 1
+			i += 1
+
+	raise PluginError("ERROR: Scene Index not found!")
+
+def getInsertionIndex(sceneNames, sceneName, index, mode):
+	'''Returns the index to know where to insert data'''
+	# special case where the scene is "Inside the Great Deku Tree"
+	# since it's the first scene simply return 0
+	if sceneName == "SCENE_YDAN":
+		return 0
```

Comment:
Why is this special case needed? I feel like `getOriginalIndex` would work just fine with deku tree

Reply:
When you export after removing the deku tree scene it will act like it's a custom scene, so I figured returning ``0`` if the scene's ``SCENE_YDAN`` was an easy solution